sapp/ui/frontend/src/Filter.js

```diff
@@ -780,7 +784,57 @@ const SaveFilterModal = (
     return <Alert type="error">{error.toString()}</Alert>;
   }
 
-  const onOk = (): void => {
+  let filterList = [];
+
+  const importProps = {
+    multiple: true,
+    onRemove: filterFile => {
+      filterList.splice(filterList.indexOf(filterFile), 1);
+    },
+    beforeUpload: filterFile => {
+      if (filterFile.type !== 'application/json') {
+        message.error(`${filterFile.name} is not a json file`);
+      } else {
+        filterList = [...filterList, filterFile];
+      }
+      return filterFile.type === 'application/json' ?
+        false : Upload.LIST_IGNORE;
+    },
+    filterList,
+  };
+
+  const onImport = () => {
+    let count = filterList.length;
+    filterList.forEach(filterFile => {
+      let reader = new FileReader();
+      reader.onload = (e) => {
+        let filter: FilterDescription = JSON.parse(e.target.result);
+        if(filter["name"] !== undefined &&
+          filter["description"] !== undefined &&
+          Object.keys(filter).length > 2
+        ) {
```

Comment:
nit: prefer early aborts, i.e. something along the lines of

```
if (!(<what we want>)) {
  // error handling, are there error dialog helpers in the React library we're using?
}
// rest of the code
```

Reply:
Yep we use message. Reusing it :)